### ISSUE TITLE
Privacy, stability etc. are not respected in imports with empty braces

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#28388]
```
use std::rt::{};
```
should warn that `std::rt` is unstable, but it doesn't.
```
mod m {
    mod n {}
}

use m::n::{};
```
should be caught by privacy checker, but it isn't.

See https://github.com/rust-lang/rust/pull/28364#issuecomment-139869943 for some details.

**Comment by petrochenkov** [comment]
In fact, almost no checks are performed on the prefix, even
```
use foo::bar::baz::{};
fn main() {}
```
compiles successfully.
I think, instead of trying to resolve the prefix we could just prohibit empty braces in imports - this construction is useless to the limit, it can't even be obtained as a result of macro expansion.

**Comment by christopherdumas** [comment]
> instead of trying to resolve the prefix we could just prohibit empty braces in imports - this construction is useless to the limit

I agree whole heartedly.

**Comment by christopherdumas** [comment]
Should I try to make empty braces disallowed? But I think that would need an RFC or at least approval from someone else.

**Comment by nagisa** [comment]
Needs RFC. There’s at least a few things impl-wise to be considered here.
